src/rpc/output_script.cpp

```diff
@@ -156,16 +156,20 @@ static RPCHelpMan createmultisig()
 
             // Make the descriptor
             std::unique_ptr<Descriptor> descriptor = InferDescriptor(GetScriptForDestination(dest), keystore);
+            const std::string descriptor_str = descriptor->ToString();
 
             UniValue result(UniValue::VOBJ);
             result.pushKV("address", EncodeDestination(dest));
             result.pushKV("redeemScript", HexStr(inner));
-            result.pushKV("descriptor", descriptor->ToString());
+            result.pushKV("descriptor", descriptor_str);
 
             UniValue warnings(UniValue::VARR);
             if (!request.params[2].isNull() && OutputTypeFromDestination(dest) != output_type) {
-                // Only warns if the user has explicitly chosen an address type we cannot generate
-                warnings.push_back("Unable to make chosen address type, please ensure no uncompressed public keys are present.");
+                // Ensure it won't cause an incorrect warning for p2sh-segwit
+                if (descriptor_str.find("wsh") == std::string::npos) {
```

Comment:
I don't particularly like using a `find` on the string to check the kind of descriptor here. It seems a bit fragile. Isn't there a more formal, well-defined way?

Reply:
`descriptor->GetOutputType()` ?